### PR TITLE
Bump go version to 1.22.1

### PR DESCRIPTION
**Pull request by chickenlj**

### CODE REVIEW

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       # If you want to matrix build , you can append the following list.
       matrix:
         go_version:
-          - 1.17
+          - 1.22.1
```
Comment (justxuewei) [inline .github/workflows/github-actions.yml:23]: No need to privode patch version.

Comment (justxuewei) [inline .github/workflows/github-actions.yml:23]: No need to privode patch version.

```suggestion
          - 1.22
```

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -18,15 +18,15 @@ jobs:
     strategy:
       matrix:
         golang:
-          - 1.17
+          - 1.22.1
```
Comment (justxuewei) [inline .github/workflows/golangci-lint.yml:21]: Ditto.

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -18,15 +18,15 @@ jobs:
...
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.22.1
```
Comment (justxuewei) [inline .github/workflows/golangci-lint.yml:25]: Ditto.

File: cmd/protoc-gen-go-triple/go.mod [hunk 4]
```diff
@@ -1,11 +1,14 @@
 module dubbo.apache.org/dubbo-go/v3/cmd/protoc-gen-go-triple
 
-go 1.15
+go 1.22.1
```
Comment (justxuewei) [inline cmd/protoc-gen-go-triple/go.mod:3]: Ditto.

File: .github/workflows/github-actions.yml (hunk 1 above)
Comment (justxuewei) [reply .github/workflows/github-actions.yml:23]: Plus, we could add latest three versions, `1.20`, `1.21` and `1.22` at this moment, on account of compatibility.

**Review by justxuewei — COMMENTED** [review]
Thanks! A few comments here...

File: .github/workflows/github-actions.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ jobs:
       # If you want to matrix build , you can append the following list.
       matrix:
         go_version:
-          - 1.17
+          - 1.20
```
Comment (DMwangnima) [inline .github/workflows/github-actions.yml:23]: There is bug in github when we configure golang version great than or equals to 1.20.
image
We should add quote on this golang version:
```
'1.20'
```

File: .github/workflows/golangci-lint.yml [hunk 6]
```diff
@@ -18,11 +18,11 @@ jobs:
     strategy:
       matrix:
         golang:
-          - 1.17
+          - 1.20
```
Comment (DMwangnima) [inline .github/workflows/golangci-lint.yml:21]: ditto.

File: .github/workflows/golangci-lint.yml [hunk 7]
```diff
@@ -18,11 +18,11 @@ jobs:
...
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.20
```
Comment (DMwangnima) [inline .github/workflows/golangci-lint.yml:25]: ditto.

**Pull request closed without merging**